### PR TITLE
Complib v0.0.3

### PR DESCRIPTION
**Pull request by Tony-Stark**
## Check List

- [ ] The title of my pull request is a short description of the changes
- [ ] This PR relates to some issue: 
- [ ] I have documented the changes made (if applicable)
- [ ] I have covered the changes with unit tests

### CODE REVIEW

File: dashboard/src/App.tsx [hunk 1]
```diff
@@ -1,13 +1,67 @@
...
+import Status from "./components/Status";
+import Button from "./components/Button";
+import RevisionCard from "./components/RevisionCard";
 function App() {
   return (
     <div>
-      <Badge type="error"> Hello </Badge>
-      <Badge type="success"> Hello </Badge>
-      
+      <hr />
```
Comment (chad11111) [inline dashboard/src/App.tsx:9]: remove all this and add to story book

File: dashboard/src/components/RevisionCard.tsx [hunk 2]
```diff
@@ -0,0 +1,141 @@
...
+      </div>
+
+      {/* if the revision was made longer than 60 seconds ago, the time is displayed in minutes */}
+      {/* if the revision was made longer than 60 minutes ago, the time is displayed in hours */}
+      {/* if the revision was made longer than 24 hours ago, the time is displayed in days */}
+      <div className="absolute bottom-0 right-0 mb-4 mr-4">
+        <span className="text-sm font-light">
+          AGE:{" "}
+          {Math.floor((Date.now() - revisionDate.getTime()) / 1000) < 60
+            ? Math.floor((Date.now() - revisionDate.getTime()) / 1000) + "s"
```
Comment (chad11111) [inline dashboard/src/components/RevisionCard.tsx:124]: extract to a function describing what is happening here.

File: dashboard/src/App.tsx [hunk 3]
```diff
@@ -1,13 +1,49 @@
...
+    The title is "Welcome to our project!"
+    The instructions include <code> and <pre> tags to display
+    the commands to run storybook and for some troubleshooting.
+    The card is bordered with slightly rounded corners.
+    The card is clickable and will redirect to localhost:6006 
+    where storybook is running.
+    If there are still any issues, the user shouldn't hesitate to ask for help.
+    no login information required. */
   return (
-    <div>
```
Comment (chad11111) [inline dashboard/src/App.tsx:5]: just remove all this from app.js

File: dashboard/src/components/RevisionCard.tsx [hunk 4]
```diff
@@ -0,0 +1,143 @@
...
+export default function RevisionCard({
+  revision,
+  revisionDate,
+  previousVersion,
+  currentVersion,
+  statusCode,
+  isActive = "false",
+  isRefreshable = false,
+  onClick
+}: {
```
Comment (chad11111) [inline dashboard/src/components/RevisionCard.tsx:76]: not urgent but for readability, you could extract the prop types to an interface declared above like 67

example:

interface Props {
}

File: dashboard/src/components/Status.tsx [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+ *
+ *
+ */
+import React from "react";
+import "../index.css";
+export default function Status({
+  statusCode,
+  isRefreshable = false
+}: {
+  statusCode: "Deployed" | "Superseded" | "Failed";
```
Comment (chad11111) [inline dashboard/src/components/Status.tsx:22]: i see that  "Deployed" | "Superseded" | "Failed" appears twice in the code base.

plz extract to type and save in general file

File: dashboard/src/components/Status.tsx [hunk 6]
```diff
@@ -0,0 +1,47 @@
...
+  isRefreshable: boolean;
+}): JSX.Element {
+  const statusVariants = {
```
Comment (chad11111) [inline dashboard/src/components/Status.tsx:25]: place content declarations on top of file  (after imports)

File: dashboard/src/components/Status.tsx [hunk 7]
```diff
@@ -0,0 +1,47 @@
...
+    Deployed: "text-green-500",
+    Superseded: "text-gray-500",
+    Failed: "text-red-500"
+  };
+  /* the text should be bold. we can use tailwind utility class for that */
+  const baseTextElem = "inline-flex font-bold";
+  // this const is used to display the refresh icon
+  const refreshIcon = (
```
Comment (chad11111) [inline dashboard/src/components/Status.tsx:33]: these constants are ok here.

File: dashboard/src/components/TabsBar.tsx [hunk 8]
```diff
@@ -0,0 +1,50 @@
...
+ *
+ * @returns {JSX.Element} - the tabs bar
+ *
+ *
+ */
+
+import React from "react";
+import "../index.css";
+
+interface TabsBarProps {
```
Comment (chad11111) [inline dashboard/src/components/TabsBar.tsx:21]: perfect do this on all components

File: dashboard/src/components/TabsBar.tsx [hunk 9]
```diff
@@ -0,0 +1,50 @@
+/**
+ * @file TabsBar.tsx
+ *
+ * This component is the bar that contains the tabs.
```
Comment (chad11111) [inline dashboard/src/components/TabsBar.tsx:4]: what is this component for ?

File: dashboard/src/components/TabsBar.tsx [hunk 10]
```diff
@@ -0,0 +1,50 @@
...
+ * @param {Function} setActiveTab - the function that should be called when a tab is clicked
+ * @param {Function} setTabContent - the function that should be called when a tab is clicked
+ *
+ * @returns {JSX.Element} - the tabs bar
+ *
+ *
+ */
+
+import React from "react";
+import "../index.css";
```
Comment (chad11111) [inline dashboard/src/components/TabsBar.tsx:19]: i think u only need to load this once in App.js

File: dashboard/src/components/RevisionCard.tsx (hunk 4 above)
Comment (Tony-Stark) [reply dashboard/src/components/RevisionCard.tsx:76]: Extracted props into interface for better readability in the next commit.

File: dashboard/src/components/Status.tsx (hunk 6 above)
Comment (Tony-Stark) [reply dashboard/src/components/Status.tsx:25]: This is declared in global.tsx now

File: dashboard/src/components/Status.tsx (hunk 5 above)
Comment (Tony-Stark) [reply dashboard/src/components/Status.tsx:22]: This is extracted to global.tsx in the next commit.

File: dashboard/src/App.tsx (hunk 3 above)
Comment (Tony-Stark) [reply dashboard/src/App.tsx:5]: Removed and changed to a storybook link

File: dashboard/src/components/TabsBar.tsx (hunk 9 above)
Comment (Tony-Stark) [reply dashboard/src/components/TabsBar.tsx:4]: This component isn't currently needed/thought through and will be deleted.

File: dashboard/src/components/TabsBar.tsx (hunk 10 above)
Comment (Tony-Stark) [reply dashboard/src/components/TabsBar.tsx:19]: It is possible, I'll try to delete it and see what happens.

File: dashboard/src/components/RevisionCard.tsx (hunk 2 above)
Comment (Tony-Stark) [reply dashboard/src/components/RevisionCard.tsx]: Extracted into getRevisionAgeJSX

File: dashboard/src/components/RevisionCard.tsx [hunk 11]
```diff
@@ -0,0 +1,152 @@
...
+  statusCode: StatusCode;
+  isActive: boolean;
+  isRefreshable: boolean;
+  onClick: () => void;
+  onRefreshClick: () => void;
+}
+
+export default function RevisionCard({
+  props
+}: {
```
Comment (chad11111) [inline dashboard/src/components/RevisionCard.tsx:91]: same as above

File: dashboard/src/components/TextInput.tsx [hunk 12]
```diff
@@ -0,0 +1,28 @@
...
+  isMandatory?: boolean;
+  onChange: (event: React.ChangeEvent<HTMLInputElement>) => void;
+}
+const defaultProps = {
+  label:'Label',
+  placeholder:'Write something here...',
+  isMandatory:false,
+  onChange:()=>{}
+}
+export default function TextInput({props}: {props: TextInputProps}): JSX.Element {
```
[the remark this answers is not in the document]
Comment (Tony-Stark) [reply dashboard/src/components/TextInput.tsx]: changed accordingly, thanks for spotting it.

File: dashboard/src/components/RevisionCard.tsx (hunk 11 above)
Comment (Tony-Stark) [reply dashboard/src/components/RevisionCard.tsx:91]: fixed

File: dashboard/src/components/Badge.tsx [hunk 13]
```diff
@@ -20,29 +20,30 @@
...
+export interface BadgeProps extends React.HTMLAttributes<HTMLSpanElement> {
+  type: 'error' | 'success' | 'info' | 'default'
+  children: React.ReactNode
+}
 export default function Badge (
-  { type, children }:
-  { type: 'error' | 'success' | 'info' | 'default'
-    children: React.ReactNode }): JSX.Element {
+  { props }:
+  { props: BadgeProps}): JSX.Element {
```
[the remark this answers is not in the document]
Comment (Tony-Stark) [reply dashboard/src/components/Badge.tsx:29]: Fixed

File: dashboard/src/components/Status.tsx [hunk 14]
```diff
@@ -0,0 +1,46 @@
...
+ * @param statusCode: enum StatusCode
+ *
+ *
+ */
+import React from "react";
+export default function Status({
+  statusCode,
+  isRefreshable = false
+}: {
+  statusCode: "Deployed" | "Superseded" | "Failed";
```
[the remark this answers is not in the document]
Comment (Tony-Stark) [reply dashboard/src/components/Status.tsx]: Fixed and added interface

File: dashboard/src/components/Status.tsx [hunk 15]
```diff
@@ -0,0 +1,48 @@
...
+ * 2. Superseded
+ * 3. Failed
+ * The status is indicated by a green orb, a neutral grey orb , or a red orb.
+ *
+ * @param statusCode: enum StatusCode
+ *
+ *
+ */
+import React from "react";
+// import global constants from global.tsx
```
[the remark this answers is not in the document]
Comment (Tony-Stark) [reply dashboard/src/components/Status.tsx]: comment removed

File: dashboard/src/components/Status.tsx [hunk 16]
```diff
@@ -0,0 +1,48 @@
...
+import {StatusCode} from '../global';
+export default function Status({
+  statusCode,
+  isRefreshable = false
+}: {
+  statusCode: StatusCode;
```
[the remark this answers is not in the document]
Comment (Tony-Stark) [reply dashboard/src/components/Status.tsx:21]: extracted to interface

**Pull request closed without merging**